miso-web/src/main/webapp/scripts/box_visualization.js

```diff
@@ -483,122 +483,44 @@ Box.ScanDialog = function() {
   return self;
 };
 
-Box.ScanErrors = function() {
+Box.ScanDiff = function() {
   var self = new BoxVisual();
 
-  self.show = function(opts) {
-    self.scan = opts.scan;
-    self.size = opts.size;
-    self.data = opts.data;
-
-    jQuery('#dialogInfoAbove').html('<h1 class="warning">Scan Failed!</h1>'+
-                                    '<p>'+self.scan.errors.message +
-                                    '<br>Read error positions:</p><br>');
-    jQuery('#dialogInfoBelow').html('');
-    jQuery('#dialogDialog').dialog({
-      autoOpen: false,
-      width: Box.dialogWidth,
-      modal: true,
-      resizable: false,
-      position: [jQuery(window).width()/2 - Box.dialogWidth/2, 50],
-      buttons: {
-        "Rescan": function () {
-          Box.initScan();
-        },
-        "Cancel": function () {
-          jQuery('#dialogDialog').dialog('close');
-        }
+  self.show = function(results) {
+    self.results = results;
+
+    var diffs = results.diffs.map(function(d) {
+      switch (d.action) {
+        case 'added':
+          return '<li style="color:green;"><b>+</b> ' + d.modified.name + ' added to the box at position ' + d.coordinates + '</li>';
+        case 'removed':
+          return '<li style="color:red;"><b>-</b> ' + d.original.name + ' removed from ' + d.coordinates + '</li>';
+        case 'changed':
+          return '<li style="color:orange;"><b>!</b> ' + d.original.name + ' replaced by ' + d.modified.name + ' at ' + d.coordinates + '</li>';
+        default:
+          return '<li><b>?</b> Unknown change at ' + d.coordinates + '</li>';
       }
     });
-    self.create({
-      div: '#dialogVisual',
-      size: {
-        rows: self.size.rows,
-        cols: self.size.cols
-      },
-      data: self.data
-    });
-    jQuery('#updateSelected, #removeSelected, #emptySelected').prop('disabled', true).addClass('disabled');
-    jQuery('#dialogDialog').dialog('open');
-  };
 
-  self.getBoxItem = function(row, col) {
-    var pos = Box.utils.getPositionString(row, col);
-    var img = '/styles/images/tube_error.png';
-    if (self.scan.errors.type == 'Unknown Barcode') {
-      img = '/styles/images/tube_duplicate_barcode_error.png';
-    } 
+    var message = (results.errors.length == 0 ? "<h1>Scan Success! </h1>" : '<h1 class="warning">Scan Failed!</h1>') +
+        results.errors.map(function(err) { return "<p>Well " + err.coordinates + ": " + err.message + "</p>"; }).join("") +
```

Comment:
Perhaps `Position` instead of `Well`, to also cover tubes in boxes as well as wells in plates?